source/value-of.d.ts

```diff
@@ -0,0 +1,13 @@
+/**
+Creates a union of all the values of the object's keys. Optionally, a key can be supplied, which limits the keys to get the values for.
+
+@example
+```
+import {ValueOf} from 'type-fest';
+
+const value: ValueOf<[1, 2, 3]> = 1
```

Comment:
It's not a blocker, but the one gripe I have with this is that this array example has the potential to cause confusion for some developers, especially with the numbers.
```suggestion
const value: ValueOf<{ key: 3 }> = 3;
```

Reply:
Done!